### ISSUE TITLE
Add Custom Search Templates on a per Title Basis

### ISSUE THREAD

**Issue by Ravencentric** [Sonarr/Sonarr#5208]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe

Having to wait 48 hours for TVDB to update or encountering an alias not in XEM to download an anime is annoying. Yes, I can request said title to be added to XEM, which I do anyway, but it doesn't change the fact that it's a process that takes time, and also titles that are only used by AnimeBytes are not supported at all. 

### Describe the solution you'd like

Allow custom search templates on a per title basis. Something similar to what Medusa offers perhaps with more customizability (regex?). While we are at it, show all the titles being used by Sonarr and let me pick and choose which ones will be used so as to not hammer my Indexer with 6 queries per episode.

### Describe alternatives you've considered

Waiting on TVDB, requesting a title in XEM, manually grabbing it.

### Anything else?

![](https://user-images.githubusercontent.com/78981416/198120223-53a96f8f-3af0-4a99-b621-38fd99adb249.jpeg)

**Comment by bakerboy448** [comment]
Duplicate of multiple other issues...please search next time

No plans for this.

TVDb doesn't need to be updated for an alias to be added to xem  so not sure what you're talking about.

for episodes or new series - if they don't exist in TVDb it doesn't matter what can be searched as sonarr knows nothing about the object and thus there's nothing to process

---—

> and also titles that are only used by AnimeBytes are not supported at all.

that's an animebytes issue because their api is garbage and they use names that literally no one else uses and yet those aliases would be searched for everyone using the series.  they made a recent change to add the file name - which jackett supports, but by doing so jackett now no longer supports any season packs.

**Comment by markus101** [comment]
The biggest reason we rely on XEM and our own scene mappings is because it gets fixed once for everyone at the same time, we do not have plans to support custom search templates within Sonarr, at least in the foreseeable future; it's not something we're going to commit to at this time.

**Comment by Ravencentric** [comment]
> Duplicate of multiple other issues...please search next time
> 
> * [Add custom series aliases #4998](https://github.com/Sonarr/Sonarr/issues/4998)
> 
> It's been repeatedly said No plans for this.

Apologies, I did search a few keywords but missed it.

> that's an animebytes issue because their api is garbage and they use names that literally no one else uses and yet those aliases would be searched for everyone using the series. 

I know and agree it's AB's fault which to me seems like it's all the more reason to add custom search to let people handle AB on their own.

> The biggest reason we rely on XEM and our own scene mappings is because it gets fixed once for everyone at the same time.

I know and it's great but having the option to set a search template that's lacking in XEM for any reason should be an option. More control over what gets searched.

Just this season, I have encountered some shows that are broken and could have been fixed by a custom search.
- **Detective Conan: The Culprit Hanzawa** - No results
- **Mob Psycho 100 S03** - AB returns `[SubsPlease] Mob Psycho 100 III 04 [Web][MKV][h264][1080p][AAC 2.0][Softsubs (SubsPlease)]` and `[SubsPlease] Mob Psycho 100 S3 04 [Web][MKV][h264][720p][AAC 2.0][Softsubs (SubsPlease)]`, neither of which work atm, so I have to resort to Nyaa.
- **Ginga Eiyuu Densetsu: Die Neue These S4** - Similar to Mob Psycho
- **Megaton-kyuu Musashi S2** - Similar to Mob Psycho

Now obviously the blame falls on AB for their bad API and all, but atleast a feature that would let the end user fix something as trivial as the search string should be added as a fallback when all fails.
